src/_langs/en/updates/2015-06-12-Media-Source-Extensions-for-Audio.markdown

```diff
@@ -0,0 +1,373 @@
+  sourceBuffer.appendWindowStart = appendTime;
+  sourceBuffer.appendWindowEnd = appendTime + gaplessMetadata.audioDuration;
+
+  // The timestampOffset field essentially tells MediaSource where in the media
+  // timeline the data given to appendBuffer() should be placed.  I.e., if the
+  // timestampOffset is 1 second, the appended data will start 1 second into
+  // playback.
+  //
+  // MediaSource requires that the media timeline starts from time zero, so we
+  // need to ensure that the data left after filtering by the append window
+  // starts at time zero.  We'll do this by shifting all of the padding we want
+  // to discard before our append time (and thus, before our append window).
+  sourceBuffer.timestampOffset =
+      appendTime - gaplessMetadata.frontPaddingDuration;
+
+  // When appendBuffer() completes, it will fire an "updateend" event signaling
+  // that it's okay to append another segment of media.  Here, we'll chain the
+  // append for the next segment to the completion of our current append.
+  if (index == 0) {
+    sourceBuffer.addEventListener('updateend', function() {
+      if (++index < SEGMENTS) {
+        GET('sintel/sintel_' + index + '.mp3',
+            function(data) { onAudioLoaded(data, index); });
+      } else {
+        // We've loaded all available segments, so tell MediaSource there are no
+        // more buffers which will be appended.
+        mediaSource.endOfStream();
+      }
+    });
+  }
+
+  // appendBuffer() will now use the timestamp offset and append window settings
+  // to filter and timestamp the data we're appending.
+  //
+  // Note: While this demo uses very little memory, more complex use cases need
+  // to be careful about memory usage or [garbage collection](#garbage-collection) may remove ranges of
+  // media in unexpected places.
+  sourceBuffer.appendBuffer(data);
+}
+{% endhighlight %}
+
+## A Seamless Waveform
+
+Lets see what our shiny new code has accomplished by taking another look at the waveform after we've applied our append windows. Below, you can see that the silent section at the end of _[sintel_0.mp3](sintel/sintel_0.mp3)_ (in red) and the silent section at the beginning of _[sintel_1.mp3](sintel/sintel_1.mp3)_ (in blue) have been removed; leaving us with a seamless transition between segments.
+
+<div class="waveform-container">
+
+<p style="text-align: center;">
+  <img src="/web/updates/images/2015-06-12-media-source-extensions-for-audio/mp3_mid.png" alt="Joining of sintel_0.mp3 and sintel_1.mp3" style="max-width: 60%; height: auto;">
+</p>
+
+## Conclusion
+
+With that we've stitched all five segments seamlessly into one and have subsequently reached the end of our demo. Before we go, you may have noticed that our _onAudioLoaded()_ method has no consideration for containers or codecs... Which means all of these techniques will work irrespective of the container or codec type; below you can replay the original demo DASH-ready fragmented MP4 instead of MP3.
+
```

Comment:
Also, I'm not sure that's an appropriate use of a semi-colon in the last line. Consider making two sentences.